### ISSUE TITLE
Feature Request & Praise

### ISSUE THREAD

**Issue by pixeloution** [webextensions/live-css-editor#86]
**Thank you** for having written this extension in the first place. It's replaced "Live CSS Editor" for me and is a vast improvement. Just in a week of usage this tool has easily saved me 2-3 hours of my time.

There are a few things I'd love to see added, if you're taking feature requests:

- **Code Folding**
    - Built into CodeMirror, but no idea if it would be difficult to enable
- **Local SASS/LESS Parsing**
    - Having the parser local to the addon would solve some security concerns
- **Debounced Auto-Compile**
    - Having errors pop up that sometime block what you're typing can be a minor annoyance - maybe have a setting to only compile after the user hasn't typed for a configurable amount of milliseconds

And if you're not taking feature requests, then I just wanted to say Thanks for the work!

**Comment by webextensions** [comment]
@pixeloution 
Thank you for your comments and feature requests :smile:

All your suggestions should be implemented at some point of time.

* **Code Folding** (**ETA:** In coming months)
  * It should be possible to add. If it turns out to be a quick change, we will attempt that. If it seems to take more time (potentially, because same editor instance is used for switching between different syntaxes), then we will postpone it and attempt again after upgrading to CodeMirror version 6. Note 1: Upgrading CodeMirror may also be a time-taking job because some of the library code is customized for features like auto-complete. Note 2: When implemented, the "gutter" (space) for the expand/collapse buttons would be hidden in default configuration to minimize the usage of the screen real-estate.

* **Local SASS/LESS Parsing** (**ETA:** In coming weeks/months, anytime before June 2023)
  * That feature is sort of already available (it had to be used in the Firefox version of the extension as per the rules of the add-ons store). The primary reason to not add that in Chrome was that it would increase the size of the extension by a few megabytes (didn't want users to worry about the extension size before installing). That reason isn't applicable anymore since the extension is well established now. Adding that (minified code) into the extension will cause a one-time hurdle of extension approval, that is the other reason. Anyways, this change has to be done (as late as June 2023) since Chrome would stop supporting remote script loading for Extension's manifest v3 (https://developer.chrome.com/docs/extensions/mv3/mv2-sunset/). And upgrading to Extension's manifest v3 is blocked by some other security concerns (related to optional permissions) and browser usage related reasons.

* **Debounced Auto-Compile** (**ETA:** In coming months/years)
  * Yes. That makes good sense (**particularly for error notifications**). A brief note about potential complexity: We may not wish to debounce the auto-compile, but rather just delay/debounce the error notifications part. Since the code has been developed and updated over the years in patches, it also awaits refactoring around the notifications implementation. We will try to improve the balance of UX and code complexity.

**Comment by webextensions** [comment]
@pixeloution 
"Local" Sass would be available along with various other updates which will be included with the release of https://github.com/webextensions/live-css-editor/releases/tag/v8.21.2 once it becomes available in coming days/weeks (v8.21.2 has been submitted to Chrome Web Store).
